src/client/bb-client.js

```diff
@@ -149,6 +149,44 @@ function completeCodeFlow(params){
   return ret.promise;
 }
 
+/**
+ * This code is needed for the page refresh/reload workflow.
+ * When the access token is nearing expriration or is expired,
+ * this function will make an ajax POST call to obtain a new
+ * access token using the current refresh token.
+ * @return promise object
+ */
+function completeTokenRefreshFlow() {
+  var ret = Adapter.get().defer();
+  var tokenResponse = getPreviousToken();
+  var state = JSON.parse(sessionStorage[tokenResponse.state]);
+  var refresh_token = tokenResponse.refresh_token;
+
+  Adapter.get().http({
+    method: 'POST',
+    url: state.provider.oauth2.token_uri,
+    data: {
+      grant_type: 'refresh_token',
+      refresh_token: refresh_token
+    },
+  }).then(function(authz) {
+    authz = $.extend(tokenResponse, authz);
+    ret.resolve(authz);
+  }, function() {
+    console.warn('Failed to exchange refresh token for access token. Please close ' +
+      'and re-launch the application again.', {
+      status: arguments[0].status,
+      statusText: arguments[0].statusText,
+      error: arguments[0].responseJSON.error,
+      error_uri: arguments[0].responseJSON.error_uri
+    });
+    ret.reject('Failed to exchange refresh token for access token. '
+      + arguments[0].responseText);
```

Comment:
The user would need to open up the console to see it.  Additionally, you can't access the console when the page is loaded within the embedded IE browser.  Am I missing something?  The thought is to set a generic message in the reject so that if the app choses to display this message, it will be something that the user can act on.